internal/aws/cwlogs/utils.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cwlogs // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/cwlogs"
+
+import (
+	"fmt"
+	"regexp"
+)
+
+// Added function to check if value is an accepted number of log retention days
+func IsValidRetentionValue(input int64) bool {
+	switch input {
+	case
+		0,
+		1,
+		3,
+		5,
+		7,
+		14,
+		30,
+		60,
+		90,
+		120,
+		150,
+		180,
+		365,
+		400,
+		545,
+		731,
+		1827,
+		2192,
+		2557,
+		2922,
+		3288,
+		3653:
+		return true
+	}
+	return false
+}
+
+// Check if the tags input is valid
+func ValidateTagsInput(input map[string]*string) error {
+	if len(input) > 50 {
+		return fmt.Errorf("invalid amount of items. Please input at most 50 tags")
+	}
+	validKeyPattern := regexp.MustCompile(`^([\p{L}\p{Z}\p{N}_.:/=+\-@]+)$`)
+	validValuePattern := regexp.MustCompile(`^([\p{L}\p{Z}\p{N}_.:/=+\-@]*)$`)
```

Comment:
Where do these regex come from?  The `\p{...}` form is uncommon and will likely have to be investigated by anyone reading this while `\w\s` would be much more legible.  See https://regex101.com/r/msjV4h/1

```suggestion
	validKeyPattern := regexp.MustCompile(`^([\w\s.:/=+\-@]+)$`)
	validValuePattern := regexp.MustCompile(`^([\w\s.:/=+\-@]*)$`)
```

Reply:
A link to that documentation should be included at least here and probably also in the README for the components that use this.  I expect that form to be confusing to most users.  You may also want to include a human-readable description of the rules, which, as I understand them, are "alphanumerics, whitespace, and `_.:/=+-!`".

Reply:
I added the link to the documentation as well as the translation for easier understanding in the code as well as the README.